tests/samplers_tests/test_nsga2.py

```diff
@@ -477,3 +478,117 @@ def test_call_after_trial_of_random_sampler() -> None:
+        NSGAIISampler(population_size=2, crossover="blxalpha"),
+        NSGAIISampler(population_size=2, crossover="sbx"),
+        NSGAIISampler(population_size=2, crossover="vsbx"),
+        NSGAIISampler(population_size=3, crossover="undx"),
+        NSGAIISampler(population_size=3, crossover="spx"),
+    ],
+)
+
+parametrize_raise_population_nsga2_sampler = pytest.mark.parametrize(
+    "crossover_name", ["undx", "undxm", "spx"]
+)
+
+
+@parametrize_nsga2_sampler
+@pytest.mark.parametrize("n_objectives", [1, 2, 3])
+def test_crossover_objectives(n_objectives: int, sampler_class: Optional[BaseSampler]) -> None:
+    n_trials = 8
+
+    study = optuna.create_study(directions=["minimize"] * n_objectives, sampler=sampler_class)
+    study.optimize(
+        lambda t: [t.suggest_float(f"x{i}", 0, 1) for i in range(n_objectives)], n_trials=n_trials
+    )
+
+    assert len(study.trials) == n_trials
+
+
+@parametrize_nsga2_sampler
+@pytest.mark.parametrize("dims", [1, 2, 3])
+def test_crossover_dims(dims: int, sampler_class: Optional[BaseSampler]) -> None:
+    def objective1(trial: optuna.Trial) -> float:
+        xs = [trial.suggest_float(f"x{dim}", -10, 10) for dim in range(dims)]
+        return sum(xs)
+
+    n_objectives = 1
+    n_trials = 8
+
+    study = optuna.create_study(directions=["minimize"] * n_objectives, sampler=sampler_class)
+    study.optimize(objective1, n_trials=n_trials)
+
+    assert len(study.trials) == n_trials
+
+
+@pytest.mark.parametrize("n_objectives", [1, 2, 3])
+def test_undxm_objective(n_objectives: int) -> None:
+    def objectivem(trial: optuna.Trial) -> List[float]:
+        xs = [trial.suggest_float(f"x{dim}", -10, 10) for dim in range(3)]
+        return xs[:n_objectives]
+
+    n_objectives = 1
+    n_trials = 8
+
+    sampler = NSGAIISampler(population_size=4, crossover="undxm")
+    study = optuna.create_study(directions=["minimize"] * n_objectives, sampler=sampler)
+    study.optimize(objectivem, n_trials=n_trials)
+
+    assert len(study.trials) == n_trials
+
+
+def test_undxm_dim() -> None:
+    def objective1_dim2(trial: optuna.Trial) -> float:
+        xs = [trial.suggest_float(f"x{dim}", -10, 10) for dim in range(2)]
+        return sum(xs)
+
+    def objective1_dim3(trial: optuna.Trial) -> float:
+        xs = [trial.suggest_float(f"x{dim}", -10, 10) for dim in range(3)]
+        return sum(xs)
+
+    n_objectives = 1
+    n_trials = 8
+
+    sampler = NSGAIISampler(population_size=4, crossover="undxm")
+    study = optuna.create_study(directions=["minimize"] * n_objectives, sampler=sampler)
+    study.optimize(objective1_dim3, n_trials=n_trials)
+
+    assert len(study.trials) == n_trials
+
+    with pytest.raises(ValueError):
+        sampler = NSGAIISampler(population_size=2, crossover="undxm")
+        study = optuna.create_study(directions=["minimize"] * n_objectives, sampler=sampler)
+        study.optimize(objective1_dim2, n_trials=n_trials)
```

Comment:
Since positive ones have not been tested for undxm (The other crossovers are all positive `@pytest.mark.parametrize("n_params", [1, 2, 3])`.), we will separate them into negative and positive ones.